deployment/databricks-connect.qmd

```diff
@@ -644,62 +637,94 @@ sc <- spark_connect(
-Connect which environment to recreate.
+The `deploy_databricks()` function makes it easier to deploy your
+content to Posit Connect. It does its best to gather all of the pieces
+of information mentioned above, and builds the correct command to
+publish.
 
-Additionally, to avoid changing the code we used to create the document,
-we want to share with Posit Connect the contents of our Databricks
-environment variables.
+The path to your content is the last piece of information we need.
+Ideally, your content is either located in its own folder inside your
+project, or it is at the root level of your project. There are three
+ways that you can let `deploy_databricks()` the path to use:
 
-The `deployApp()` function from the `rsconnect` package allows us to
-easily do all this:
+-   If you are in RStudio, and the document you wish to publish is open,
+    `deploy_databricks()` will use the RStudio's API to get the path of
+    that document, and then use its containing folder as the content's
+    location. Our hope is that for most time, this will be how you
+    deploy.
 
-``` r
-rsconnect::deployApp(
-  python = "Enter the Python environment to recreate",
-  envVars = c("DATABRICKS_HOST", "DATABRICKS_TOKEN"),
-  lint = FALSE
-)
-```
+-   Use the `appDir` argument to pass the folder path to be used.
+    Something such as `here::here("my-cool-document")` would work.
+
+-   If no document is opened in RStudio, and `appDir` is left empty,
+    then whatever `getwd()` returns is what will be used
+
+The aim of of the new function is to be as flexible, and helpful. It
+will gather your document location, credentials, and URLs. The
+**Python** location is what you will need to define at the time of
+calling `deploy_databricks()`. If you are in RStudio, and your document
+is opened, here are several ways to use:
+
+-   If you know the cluster's DBR version:
+
+    ``` r
+    pysparklyr::deploy_databricks(version = "14.1")
+    ```
 
-In the `python` argument, paste the value from the 'Path:' output
-`spark_connect()` used when you were testing your document locally. The
-`envVars` argument will let Posit Connect know to create those
-environment variables and to use the values from your machine.
+-   The cluster's ID can also be used, so `pysparklyr` pulls the DBR
+    version:
+
+    ``` r
+    pysparklyr::deploy_databricks(cluster_id = "1026-175310-7cpsh3g8")
+    ```
+
+-   If you just ran the code of the content you plan to deploy, then the
+    **Python** environment should already be loaded in the R session.
+    The function will validate that the path of that Python environment
+    conforms to one that `pysparklyr` creates, and uses that. This will
+    happen if no `version`, `cluster_id`, or `python` is passed. At that
+    point simply run:
+
+    ``` r
+    pysparklyr::deploy_databricks()
+    ```
+
+-   You can also pass the path to the Python environment to use by
+    setting the `python` argument:
```

Comment:
I think this should be rearranged slightly so that the `python` argument is mentioned before the previous statement since it references that argument:
>This will happen if no `version`, `cluster_id`, or `python` argument is provided. At that point simply run:

Reply:
Yeah, I struggled with sequencing this one, it doesn't look right either way. But I'll give it more thought. Thanks